lib/note-content-editor.jsx

```diff
@@ -1,11 +1,68 @@
 import React, { PropTypes } from 'react';
-import { Editor, EditorState, ContentState } from 'draft-js';
+import {
+	ContentState,
+	Editor,
+	EditorState,
+	Modifier,
+} from 'draft-js';
 import { invoke, noop } from 'lodash';
 
 function plainTextContent( editorState ) {
 	return editorState.getCurrentContent().getPlainText( '\n' )
 }
 
+// contains methods that mirror the methods on Modifier with 2 differences:
+// 1. operate on EditorState instances
+// 2. move selection to the natural place following the modification
+const EditorStateModifier = {
+	insertText( editorState, selection, characters ) {
+		// this function assumes selection is collapsed
```

Comment:
importantly though, `insertText` _will_ overwrite a non-collapsed section in production - just throws an error when it does

it would be nice here to either mimic that behavior or simply remove the note about assuming. if we send in a selection I think the general assumption is that things will write over it.

this brings me back to wondering about the utility of these functions attempting to mimic `draft-js`. we're clearly _not_ trying to match behavior by means of messing with the selection. we could do ourselves well by breaking that notion and asserting "this is _our_ insert text function"